Traceback (most recent call last):
  File "/home/tbent/.miniconda3/envs/bie_book/lib/python3.9/site-packages/jupyter_cache/executors/utils.py", line 51, in single_nb_execution
    executenb(
  File "/home/tbent/.miniconda3/envs/bie_book/lib/python3.9/site-packages/nbclient/client.py", line 1112, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/home/tbent/.miniconda3/envs/bie_book/lib/python3.9/site-packages/nbclient/util.py", line 74, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/home/tbent/.miniconda3/envs/bie_book/lib/python3.9/site-packages/nbclient/util.py", line 53, in just_run
    return loop.run_until_complete(coro)
  File "/home/tbent/.miniconda3/envs/bie_book/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/home/tbent/.miniconda3/envs/bie_book/lib/python3.9/site-packages/nbclient/client.py", line 553, in async_execute
    await self.async_execute_cell(
  File "/home/tbent/.miniconda3/envs/bie_book/lib/python3.9/site-packages/nbclient/client.py", line 857, in async_execute_cell
    self._check_raise_for_error(cell, exec_reply)
  File "/home/tbent/.miniconda3/envs/bie_book/lib/python3.9/site-packages/nbclient/client.py", line 760, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
import numpy as np
import matplotlib.pyplot as plt
import matplotlib.patches as patches

%config InlineBackend.figure_format='retina'

plt.figure(figsize=(8,4))


plt.subplot(1,2,1)
theta = np.linspace(0, 2*np.pi, 500)
xsS = np.cos(theta) * (1.0 + 0.3 * np.sin(theta * 5))
ysS = np.sin(theta) * (1.0 + 0.3 * np.sin(theta * 5))

dxdt = np.sin(t) * (-0.3 * np.sin(5 * t) - 1) + 1.5 * np.cos(t) * np.cos(5 * t)
dydt = np.cos(t) + 1.5 * cos(5 * t) * np.sin(t) + 0.3 * np.cos(t) * np.sin(5 * t)
ddt_norm = np.sqrt(dxdt ** 2 + dydt ** 2)
dxdt /= ddt_norm
dydt /= ddt_norm
nx = dydt
ny = -dxdt
jacobian = 2 * np.pi * ddt_norm

plt.plot(xsS, ysS, 'k-')
plt.text(-0.95, -0.5, "$S$", fontsize=30)
plt.gca().add_patch(patches.Rectangle((0.6,0.4), 0.6, 0.6, edgecolor='k', linestyle='--', facecolor='none'))
plt.axis('off')

theta_C_idx = 100
theta_C = theta_C[theta_C_idx]
xsS_C = xsS[theta_C_idx]
ysS_C = xsS[theta_C_idx]
Cx = xsS_C + 0.1 * nx[theta_C_idx]
Cy = ysS_C + 0.1 * ny[theta_C_idx]
plt.plot([Cx], [Cy], 'bo')

plt.subplot(1,2,2)
plt.plot(xsS, ysS, 'k-')
plt.gca().add_patch(patches.Rectangle((0.6,0.4), 0.6, 0.6, edgecolor='k', linestyle='--', facecolor='none'))
plt.text(0.62, 0.95, 'Zoom', fontsize=20)
plt.text(0.62, 0.6, '$S$', fontsize=20)
plt.xlim([0.6, 1.2])
plt.ylim([0.4, 1.0])
plt.axis('off')

plt.tight_layout()
plt.show()
# circle = plt.Circle((0,0), node.radius, fill=False, color='b' if node.is_leaf else 'k')
# plt.gca().add_patch(circle)
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mNameError[0m                                 Traceback (most recent call last)
[0;32m<ipython-input-1-8f324287b70b>[0m in [0;36m<module>[0;34m[0m
[1;32m     13[0m [0mysS[0m [0;34m=[0m [0mnp[0m[0;34m.[0m[0msin[0m[0;34m([0m[0mtheta[0m[0;34m)[0m [0;34m*[0m [0;34m([0m[0;36m1.0[0m [0;34m+[0m [0;36m0.3[0m [0;34m*[0m [0mnp[0m[0;34m.[0m[0msin[0m[0;34m([0m[0mtheta[0m [0;34m*[0m [0;36m5[0m[0;34m)[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[1;32m     14[0m [0;34m[0m[0m
[0;32m---> 15[0;31m [0mdxdt[0m [0;34m=[0m [0mnp[0m[0;34m.[0m[0msin[0m[0;34m([0m[0mt[0m[0;34m)[0m [0;34m*[0m [0;34m([0m[0;34m-[0m[0;36m0.3[0m [0;34m*[0m [0mnp[0m[0;34m.[0m[0msin[0m[0;34m([0m[0;36m5[0m [0;34m*[0m [0mt[0m[0;34m)[0m [0;34m-[0m [0;36m1[0m[0;34m)[0m [0;34m+[0m [0;36m1.5[0m [0;34m*[0m [0mnp[0m[0;34m.[0m[0mcos[0m[0;34m([0m[0mt[0m[0;34m)[0m [0;34m*[0m [0mnp[0m[0;34m.[0m[0mcos[0m[0;34m([0m[0;36m5[0m [0;34m*[0m [0mt[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m     16[0m [0mdydt[0m [0;34m=[0m [0mnp[0m[0;34m.[0m[0mcos[0m[0;34m([0m[0mt[0m[0;34m)[0m [0;34m+[0m [0;36m1.5[0m [0;34m*[0m [0mcos[0m[0;34m([0m[0;36m5[0m [0;34m*[0m [0mt[0m[0;34m)[0m [0;34m*[0m [0mnp[0m[0;34m.[0m[0msin[0m[0;34m([0m[0mt[0m[0;34m)[0m [0;34m+[0m [0;36m0.3[0m [0;34m*[0m [0mnp[0m[0;34m.[0m[0mcos[0m[0;34m([0m[0mt[0m[0;34m)[0m [0;34m*[0m [0mnp[0m[0;34m.[0m[0msin[0m[0;34m([0m[0;36m5[0m [0;34m*[0m [0mt[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[1;32m     17[0m [0mddt_norm[0m [0;34m=[0m [0mnp[0m[0;34m.[0m[0msqrt[0m[0;34m([0m[0mdxdt[0m [0;34m**[0m [0;36m2[0m [0;34m+[0m [0mdydt[0m [0;34m**[0m [0;36m2[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m

[0;31mNameError[0m: name 't' is not defined
NameError: name 't' is not defined

